File: docs-examples/example-python/src/main/python/micronaut/configuration/openapi/docs/filters/OpenApiViewCookieContextPathFilter.py
from typing import Annotated

# tag::imports[]
from java.time import Duration
from micronaut.context.annotation import Requires, Value
from micronaut.core.async_.publisher import Publishers
from micronaut.http import HttpMethod, HttpRequest, MutableHttpResponse
from micronaut.http.annotation import Filter
from micronaut.http.cookie import Cookie
from micronaut.http.filter import HttpServerFilter, ServerFilterChain
from org.reactivestreams import Publisher
# end::imports[]


# tag::clazz[]
@Requires(property="micronaut.server.context-path")
@Filter(methods=[HttpMethod.GET, HttpMethod.HEAD], patterns=["/**/rapidoc*", "/**/redoc*", "/**/swagger-ui*", "/**/openapi-explorer*"])
class OpenApiViewCookieContextPathFilter(HttpServerFilter):

    def __init__(self, contextPath: Annotated[str, Value("${micronaut.server.context-path}")]):
        self.contextPathCookie = Cookie.of("contextPath", contextPath).maxAge(Duration.ofMinutes(2))

    def doFilter(self, request: HttpRequest, chain: ServerFilterChain) -> Publisher[MutableHttpResponse]:
        return Publishers.map(chain.proceed(request), lambda response: response.cookie(self.contextPathCookie))
# end::clazz[]

File: docs-examples/example-kotlin/src/main/kotlin/io/micronaut/configuration/openapi/docs/filters/OpenApiViewCookieContextPathFilter.kt
package io.micronaut.configuration.openapi.docs.filters

// tag::imports[]
import io.micronaut.context.annotation.Requires
import io.micronaut.context.annotation.Value
import io.micronaut.core.async.publisher.Publishers
import io.micronaut.http.HttpMethod
import io.micronaut.http.HttpRequest
import io.micronaut.http.MutableHttpResponse
import io.micronaut.http.annotation.Filter
import io.micronaut.http.cookie.Cookie
import io.micronaut.http.filter.HttpServerFilter
import io.micronaut.http.filter.ServerFilterChain
import org.reactivestreams.Publisher
import java.time.Duration
// end::imports[]

// tag::clazz[]
@Requires(property = "micronaut.server.context-path")
@Filter(methods = [HttpMethod.GET, HttpMethod.HEAD], patterns = ["/**/rapidoc*", "/**/redoc*", "/**/swagger-ui*", "/**/openapi-explorer*"])
class OpenApiViewCookieContextPathFilter(@Value("\${micronaut.server.context-path}") contextPath: String) : HttpServerFilter {

    private val contextPathCookie: Cookie = Cookie.of("contextPath", contextPath).maxAge(Duration.ofMinutes(2L))

    override fun doFilter(request: HttpRequest<*>, chain: ServerFilterChain): Publisher<MutableHttpResponse<*>> {
        return Publishers.map(chain.proceed(request)) { response -> response.cookie(contextPathCookie) }
    }
}
// end::clazz[]
